apps/04TowersOfHanoi.js

```diff
@@ -3,91 +3,170 @@
 var assert = require('assert');
 var readline = require('readline');
 var rl = readline.createInterface({
-  input: process.stdin,
-  output: process.stdout
+    input: process.stdin,
+    output: process.stdout
 });
 
+var num = 0;
 var stacks = {
-  a: [4, 3, 2, 1],
-  b: [],
-  c: []
+    a: [4,3,2,1],
+    b: [],
+    c: []
 };
 
 function printStacks() {
-  console.log("a: " + stacks.a);
-  console.log("b: " + stacks.b);
-  console.log("c: " + stacks.c);
+    console.log("a: " + stacks.a);
+    console.log("b: " + stacks.b);
+    console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
+function movePiece(startStack, endStack) {
+    // Your code here
+
+    //check for legal move first.
+    var answer = isLegal(startStack, endStack);
+    //if return is NOT false then pop/push
+      if (answer != false) {
```

Comment:
You can shorthand your code by passing in the variable by itself in your `if` statement (line 29)

Also, only use `===` or `!==` when making comparisons.